beacon_node/network/src/subnet_service/mod.rs

```diff
@@ -485,7 +485,7 @@ impl<T: BeaconChainTypes> SubnetService<T> {
         } else {
             // This is a future slot, schedule subscribing.
             self.scheduled_subscriptions
-                .insert_at(subnet, time_to_subscription_start);
+                .insert_at(ExactSubnet { subnet, slot }, time_to_subscription_start);
```

Comment:
Nice! We did notice a potential regression here, I looked back over this code and didn't see it. Thanks for catching this!

Could we add a comment above here, saying that we need to include the slot to make the key unique to prevent overwriting the entry for the same subnet (to prevent future people like me from removing the exact subnet here) :)

Reply:
Added the comment